### PR TITLE
Browser drag n drop target support + new DataFormat.Files which returns IStorageItem collection

### PR DESCRIPTION
**Pull request by maxkatz6**
## What does the pull request do?

Implements support for drag n drop target for the browser backend. Note, you still can't drag n drop from the browser to the outside, as our current API is rather limiting.

## Breaking changes

Some specific methods were moved from the IDataObject to the extensions methods:
- string[] GetFileNames()
- string GetText()

## Obsoletions / Deprecations

DataFormats.FileNames is obsolete now. Please use DataFormats.Files and use collection of IStorageItem instead of collection of strings.

https://user-images.githubusercontent.com/3163374/219842949-f4bb8c2e-0557-4066-88e0-01492dff6904.mp4

### CODE REVIEW

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0030234-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0030238-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

File: src/Browser/Avalonia.Browser/webapp/modules/avalonia/generalHelpers.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+        }
+
+        const retItems = [];
+        for (let i = 0; i < items.length; i++) {
+            retItems[i] = items[i];
+        }
+        return retItems;
+    }
+
+    public static callMethod(instance: any, name: string /*, args */): any {
```
Comment (emmauss) [inline src/Browser/Avalonia.Browser/webapp/modules/avalonia/generalHelpers.ts:15]: The commented args can be removed.

**Review by emmauss — CHANGES_REQUESTED** [review]

File: src/Browser/Avalonia.Browser/webapp/modules/avalonia/generalHelpers.ts (hunk 1 above)
Comment (maxkatz6) [reply src/Browser/Avalonia.Browser/webapp/modules/avalonia/generalHelpers.ts:15]: It's just shows, that we do use arguments. But it's not mapped on the actual typescript arguments list. 
It works similarly to "params" in c#, where we specify first two arguments, and everything else is read from special magic "arguments" property.

**Review by emmauss — APPROVED** [review]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0030343-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0030376-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

**Pull request merged**